src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -52,67 +76,176 @@ public void AddDependencyLocation(string fullPath)
                 {
                     _knownAssemblyPathsBySimpleName[simpleName] = paths.Add(fullPath);
                 }
+
+                // Ensure that there is no cached Assembly information about this location. Long 
+                // lived processes like VS and VBCSCompiler will see the same location added as
+                // a dependency many times. Each time have to assume there is new content on disk
+                // that needs to be considered.
+                _analyzerAssemblyInfoMap[fullPath] = null;
             }
         }
 
-        public Assembly LoadFromPath(string fullPath)
+        public Assembly LoadFromPath(string originalAnalyzerPath)
         {
-            CompilerPathUtilities.RequireAbsolutePath(fullPath, nameof(fullPath));
-            return LoadFromPathUnchecked(fullPath);
-        }
+            CompilerPathUtilities.RequireAbsolutePath(originalAnalyzerPath, nameof(originalAnalyzerPath));
 
-        #endregion
+            (AssemblyName? assemblyName, string _) = GetAssemblyInfoForPath(originalAnalyzerPath);
+
+            // Not a managed assembly, nothing else to do
+            if (assemblyName is null)
+            {
+                throw new ArgumentException("Not a valid assembly");
+            }
+
+            try
+            {
+                return Load(assemblyName, originalAnalyzerPath);
```

Comment:
The `originalAssemblyPath` is really only used to partition out the analyzer dependencies into `DirectoryLoadContext`. It really has no other usage. You can see that in the `.Desktop` case the variable is named `_` to be explicit about it not being used. 

One subtle detail here that is hard to doc / explain is that `AssemblyName.Location` has the real path in it for .NET Framework. That is really key to making the entire load scenario work on .NET Framework. For .NET Core though the strategy is a bit different because we have directory based isolation.